src/order/locally_finite.lean

```diff
@@ -482,6 +482,154 @@ locally_finite_order.of_Icc' (α × β)
+    | (a : α), (b : α) := (Ioo a b).map embedding.coe_with_top
+    end,
+  finset_mem_Icc := λ a b x, match a, b, x with
+    |       ⊤,       ⊤,       x := mem_singleton.trans (le_antisymm_iff.trans $ and_comm _ _)
+    |       ⊤, (b : α),       x := iff_of_false (not_mem_empty _)
+                                     (λ h, (h.1.trans h.2).not_lt $ coe_lt_top _)
+    | (a : α),       ⊤,       ⊤ := by simp [with_top.locally_finite_order._match_1]
+    | (a : α),       ⊤, (x : α) := by simp [with_top.locally_finite_order._match_1, coe_eq_coe]
+    | (a : α), (b : α),       ⊤ := by simp [with_top.locally_finite_order._match_1]
+    | (a : α), (b : α), (x : α) := by simp [with_top.locally_finite_order._match_1, coe_eq_coe]
+    end,
+  finset_mem_Ico := λ a b x, match a, b, x with
+    |       ⊤,       b,       x := iff_of_false (not_mem_empty _)
+                                     (λ h, not_top_lt $ h.1.trans_lt h.2)
+    | (a : α),       ⊤,       ⊤ := by simp [with_top.locally_finite_order._match_3]
+    | (a : α),       ⊤, (x : α) := by simp [with_top.locally_finite_order._match_3, coe_eq_coe,
+                                        coe_lt_top]
+    | (a : α), (b : α),       ⊤ := by simp [with_top.locally_finite_order._match_3]
+    | (a : α), (b : α), (x : α) := by simp [with_top.locally_finite_order._match_3, coe_eq_coe,
+                                        coe_lt_coe]
+    end,
+  finset_mem_Ioc := λ a b x, match a, b, x with
+    |       ⊤,       b,       x := iff_of_false (not_mem_empty _)
+                                     (λ h, not_top_lt $ h.1.trans_le h.2)
+    | (a : α),       ⊤,       ⊤ := by simp [with_top.locally_finite_order._match_4, coe_lt_top]
+    | (a : α),       ⊤, (x : α) := by simp [with_top.locally_finite_order._match_4, coe_eq_coe,
+                                        coe_lt_coe]
+    | (a : α), (b : α),       ⊤ := by simp [with_top.locally_finite_order._match_4]
+    | (a : α), (b : α), (x : α) := by simp [with_top.locally_finite_order._match_4, coe_eq_coe,
+                                        coe_lt_coe]
+    end,
+  finset_mem_Ioo := λ a b x, match a, b, x with
+    |       ⊤,       b,       x := iff_of_false (not_mem_empty _)
+                                     (λ h, not_top_lt $ h.1.trans h.2)
+    | (a : α),       ⊤,       ⊤ := by simp [with_top.locally_finite_order._match_6, coe_lt_top]
+    | (a : α),       ⊤, (x : α) := by simp [with_top.locally_finite_order._match_6, coe_eq_coe,
+                                        coe_lt_coe, coe_lt_top]
+    | (a : α), (b : α),       ⊤ := by simp [with_top.locally_finite_order._match_6]
+    | (a : α), (b : α), (x : α) := by simp [with_top.locally_finite_order._match_6, coe_eq_coe,
+                                        coe_lt_coe]
+    end }
+
+end order_top
+
+section order_bot
+open with_bot
+variables (α) [partial_order α] [order_bot α] [locally_finite_order α]
+
+instance : locally_finite_order (with_bot α) :=
```

Comment:
Can `order_dual` help you here?

Reply:
Because of the way we've constructed the `order_dual.locally_finite_order` instance, I wouldn't try it. And can Lean even turn the `≤` on `order_dual (with_top α)` into the one on `with_bot α`?

Reply:
Lean accepts:
```lean
@order_dual.locally_finite_order (with_top (order_dual α)) _ _
```

Reply:
In principle that defnition must be correct, since:
* The type of `with_bot.locally_finite_order` is correct (lean isn't doing the thing where the definition affects the type)
* `locally_finite_order` is subsingleton!

Reply:
So you're telling to not care too much about definitional unfolding?

Reply:
For this amount of line savings, possibly not. Does this definition unfold to something different anyway? Do the simp lemmas you added stop being rfl?

Reply:
If you care about definitional unfolding, you can keep all the data fields, but add `..@order_dual.locally_finite_order (with_top (order_dual α)) _ _` instead of the propositional fields. I think that should work.

But with the equational lemmas below, I'm don't think we care about the precise definition too much (as long as it's definitionally equal).